src/OSPSuite.Core/Domain/PKAnalyses/DynamicPKParameter.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System.Linq;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Utility.Extensions;
+
+namespace OSPSuite.Core.Domain.PKAnalyses
+{
+   public enum StandardPKParameter
+   {
+      Unknown,
+      Cmax,
+      Cmin,
+      Tmax,
+      Tmin,
+      CTrough,
+      Auc,
+      Aucm,
+      AucInf,
+      AucTendInf,
+      Mrt,
+      FractionAucEndToInf,
+      Thalf,
+      Vss,
+      Vd,
+      Tthreshold,
+   }
+
+   public class DynamicPKParameter : PKParameter
+   {
+      public DynamicPKParameter()
+      {
+         // Default dimension for a dynamic PK Parameter for now 
+         Dimension = Constants.Dimension.NO_DIMENSION;
+      }
+
+      // Parameter such as AUC, Cmax, Tmax that will be used to calculate the parameter
+      public StandardPKParameter StandardPKParameter { get; set; } = StandardPKParameter.Unknown;
+
+      public double? StartTime { get; set; }
+      public double? EndTime { get; set; }
+
+      /// <summary>
+      ///    0-based index of the application that should be used to estimate the start of the interval
+      /// </summary>
+      public int? StartApplicationIndex { get; set; }
+
+      /// <summary>
+      ///    0-based index of the application that should be used to estimate the end of the interval
+      /// </summary>
+      public int? EndApplicationIndex { get; set; }
+
+      /// <summary>
+      ///    Time offset in min to apply to the start time (either from the absolute <see cref="StartTime" /> or the start time
+      ///    estimated from the application <see cref="StartApplicationIndex" />
+      /// </summary>
+      public double? StartTimeOffset { get; set; }
+
+      /// <summary>
+      ///    Time offset in min to apply to the start time (either from the absolute <see cref="EndTime" /> or the start time
+      ///    estimated from the application <see cref="EndApplicationIndex" />
+      /// </summary>
+      public double? EndTimeOffset { get; set; }
```

Comment:
I am returning double here because it creates issue with signature not matchng in R (double vs float)